spec/cipherstash/index/bloom_filter_spec.rb

```diff
@@ -3,27 +3,133 @@
 require "cipherstash/index/bloom_filter"
 
 describe CipherStash::Index::BloomFilter do
-  # Generated by SecureRandom.hex(16)
+  # Generated by SecureRandom.hex(32)
   # The same key is used for each test run so that these tests are deterministic.
-  let(:key) { "8b42157da439e06aa6f6d1ad8c1db72e" }
+  let(:key) { "b6d6dba3be33ffaabb83af611ec043b9270dacdc7b3015ce2c36ba17cf2d3b2c" }
 
   describe ".new" do
     it "returns a bloom filter with empty bits" do
       filter = described_class.new(key)
       expect(filter.bits).to eq(Set.new())
     end
+
+    it "provides a default for m" do
+      filter = described_class.new(key)
+      expect(filter.m).to eq(256)
+    end
+
+    described_class::VALID_M_VALUES.each do |m|
+      it "allows #{m} as a value for m" do
+        filter = described_class.new(key, {"filterSize" => m})
+        expect(filter.m).to eq(m)
+      end
+    end
+
+    [0, 2, 16, 31, 513, 131072].each do |m|
```

Comment:
```suggestion
    [0, 2, 16, 31, 513, 131072, "256", "ohai", nil, { foo: "bar" }, Object.new].each do |m|
```

I am a truly terrible person.